e2e/cypress/integration/ui/settings/03_team_add_users.spec.ts

```diff
@@ -0,0 +1,177 @@
+  const cypressPrefix = 'cypress test';
+  const nameForUser = cypressPrefix + ' user ' + now;
+  const usernameForUser = 'testing-user-' + now;
+  const descriptionForTeam = cypressPrefix + ' team ' + now;
+  const nameForTeam = 'testing-team-' + now;
+  let teamID = '';
+  let adminToken = '';
+  let teamUIRouteIdentifier = '';
+  let iamVersion = '';
+
+  before(() => {
+    cy.adminLogin('/settings/teams').then(() => {
+      // clean up leftover teams in case of previous test failures
+      const admin = JSON.parse(<string>localStorage.getItem('chef-automate-user'));
+      adminToken = admin.id_token;
+      cy.cleanupUsersByNamePrefix(adminToken, cypressPrefix);
+      cy.cleanupTeamsByDescriptionPrefix(adminToken, cypressPrefix);
+
+      // create custom user and team
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'POST',
+        url: '/api/v0/auth/users',
+        body: {
+          username: usernameForUser,
+          name: nameForUser,
+          password: 'chefautomate'
+        }
+      });
+
+      cy.request({
+        auth: { bearer: adminToken },
+        method: 'POST',
+        url: '/api/v0/auth/teams',
+        body: {
+          description: descriptionForTeam,
+          name: nameForTeam
+        }
+      }).then((resp) => {
+        teamID = resp.body.team.id;
+
+        cy.get('chef-sidebar')
+        .invoke('attr', 'major-version')
+        .then((obj: Cypress.ObjectLike) => {
+          // Cypress.ObjectLike can't be casted to a string directly,
+          // so must convert to Object type (common to all JS objects) first
+          iamVersion = <string><Object>obj;
+
+          if (iamVersion === 'v2') {
+            teamUIRouteIdentifier = nameForTeam;
+          } else {
+            teamUIRouteIdentifier = teamID;
+          }
+        });
+      });
+
+    });
+  });
+
+  beforeEach(() => {
+    cy.restoreStorage();
+
+    if (iamVersion === 'v2') {
+      cy.route('GET', `/apis/iam/v2beta/teams/${nameForTeam}`).as('getTeam');
+      cy.route('GET', `/apis/iam/v2beta/teams/${nameForTeam}/users`).as('getTeamUsers');
+      cy.route('GET', '/apis/iam/v2beta/users').as('getUsers');
+    } else {
+      cy.route('GET', `/api/v0/auth/teams/${teamID}`).as('getTeam');
+      cy.route('GET', `/api/v0/auth/teams/${teamID}/users`).as('getTeamUsers');
+      cy.route('GET', '/api/v0/auth/users').as('getUsers');
+    }
+
+    cy.visit(`/settings/teams/${teamUIRouteIdentifier}/add-users`);
+    cy.wait(['@getTeam', '@getTeamUsers', '@getUsers']);
+
+    cy.get('app-welcome-modal').invoke('hide');
+  });
+
+  afterEach(() => {
+    cy.saveStorage();
+  });
+
+  after(() => {
+    cy.cleanupUsersByNamePrefix(adminToken, cypressPrefix);
+    cy.cleanupTeamsByDescriptionPrefix(adminToken, cypressPrefix);
+  });
+
+  it('when the x is clicked, it returns to the team details page', () => {
+    cy.get('chef-page chef-button.close-button').click().then(() => {
```

Comment:
No need to nest with `then`; you only need a `then` if you are passing data, as in L136.

This same comment applies to most of the other `then` instances below, too.